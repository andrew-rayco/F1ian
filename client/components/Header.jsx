import React from 'react'

const Header = () => {

  return (
    <div className="header">
      <nav>
          <a id="home-link" href="/#/">Home</a>
      </nav>
      <h1>F1ian</h1>
    </div>
  )
}

module.exports = Header
