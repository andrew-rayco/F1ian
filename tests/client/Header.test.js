import React from 'react'
import { shallow } from 'enzyme'
import Header from '../../client/components/Header'

describe('Header', () => {
  let props = { location: {pathname:"/season/2014"}}
  const app = shallow(<Header {...props} />)

  it('renders correctly', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains an h1', () => {
    expect(app.find('h1').exists()).toBe(true)
  })

  it('contains an h1 with the correct text', () => {
    expect(app.find('h1').text()).toEqual('F1ian')
  })

  it('contains a .main-header div', () => {
    expect(app.find('.main-header').exists()).toBe(true)
  })
})
