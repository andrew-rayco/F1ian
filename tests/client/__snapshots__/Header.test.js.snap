// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    location={
      Object {
        "pathname": "/season/2014",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav>
          <a
            href="/#/"
            id="home-link"
          >
            Home
          </a>
        </nav>,
        <h1>
          F1ian
        </h1>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="/#/"
            id="home-link"
          >
            Home
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "href": "/#/",
            "id": "home-link",
          },
          "ref": null,
          "rendered": "Home",
          "type": "a",
        },
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "F1ian",
        },
        "ref": null,
        "rendered": "F1ian",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav>
            <a
              href="/#/"
              id="home-link"
            >
              Home
            </a>
          </nav>,
          <h1>
            F1ian
          </h1>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/#/"
              id="home-link"
            >
              Home
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "href": "/#/",
              "id": "home-link",
            },
            "ref": null,
            "rendered": "Home",
            "type": "a",
          },
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "F1ian",
          },
          "ref": null,
          "rendered": "F1ian",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
